theano/tensor/sort.py

```diff
@@ -80,11 +80,43 @@ def grad(self, inputs, output_grads):
                 idx = argsort(*inputs, kind=self.kind, order=self.order)
                 # not working: numpy.where(idx[None, :]==numpy.arange(2)[:, None, None])
                 pass
+        elif a.ndim == 3:
+            if isinstance(axis, theano.Constant) and axis.data is not None:
+                indices = self.__get_argsort_indices(a, axis)
+                inp_grad = output_grads[0].reshape(a.shape)[indices[0], indices[1], indices[2]]
```

Comment:
the reshape here isn't needed. output_grads[0] should be of the same shape as a.